### PR TITLE
usePopup for article page toc component

### PR DESCRIPTION
**Pull request by hueitan**

### CODE REVIEW

File: src/components/Article.js [hunk 1]
```diff
@@ -81,16 +81,16 @@ const ArticleInner = ({ lang, articleTitle, initialSubTitle }) => {
     return 'Loading...'
   }
 
-  const [isTocShown, toggleToc] = useState(false)
   const [subTitle, setSubTitle] = useState(initialSubTitle)
+  const [popupToc] = usePopup(ArticleToc)
```
Comment (stephanebisson) [inline src/components/Article.js]: For consistency, the function here should be named show<Something>Popup(). For instance `showArticleTocPopup` or just `showTocPopup`

File: src/components/ArticleToc.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { useRef, useEffect } from 'preact/hooks'
 import { useNavigation, useI18n, useSoftkey } from 'hooks'
 import { ListView } from 'components'
 
-export const ArticleToc = ({ items, close }) => {
+export const ArticleToc = ({ items, closeFn, close }) => {
```
Comment (stephanebisson) [inline src/components/ArticleToc.js]: We can probably find a better name for `closeFn`. Maybe `onSelectItem`...

File: src/components/Article.js (hunk 1 above)
Comment (hueitan) [reply src/components/Article.js]: see if it is better now

File: style/toc.less [hunk 3]
```diff
@@ -13,7 +13,7 @@
 
 	.list {
 		top: @pageHeaderHeight;
-		bottom: @softkeysHeight;
+		bottom: 0px;
```
Comment (stephanebisson) [inline style/toc.less]: 0 should be just 0 with no units.

**Pull request merged**